torch/jit/__init__.py

```diff
@@ -1602,6 +1602,25 @@ def __init__(self, original, stubs):
                             continue
                     ScriptModule.__setattr__(self, name, item)
 
+            # Copy annotations, pull types from `__annotations__` or try to infer
+            # the type if possible
+            class_annotations = getattr(original, '__annotations__', {})
+            for name in dir(original):
```

Comment:
doesn't dir(value) gets a ton of arbitrary python implementation details ? these are all nn.Modules, we could try register a hook in __setattr__ or something else

Reply:
As far as I know we don't really have any other choice since things like `__dict__` only have the current class' info (not anything from superclasses), plus not everything goes through `__setattr__` (e.g. static members). This is mostly safe since the only things it takes are types it can infer, so it would pick up the `__module__` and `__doc__`

Reply:
Looks like it would also pick up `_version` `inplace` `dump_patches`

Reply:
why doens't it work if you add it here ? https://github.com/pytorch/pytorch/blob/c25e33789e45e20d3fa317f60e29bdbe49bffd28/torch/nn/modules/module.py#L584

Reply:
That'd be mostly fine but if something is defined at the class level it doesn't go through `__setattr__`, it's just on the class type itself, we need to copy those too (including any up the inheritance chain) so it doesn't work generally